dask/dataframe/io/delta/core.py

```diff
@@ -0,0 +1,289 @@
+import json
+import re
+
+from fsspec.core import get_fs_token_paths
+from pyarrow import dataset as pa_ds
+
+from ....delayed import delayed
+from ..io import from_delayed
+from ..parquet.core import get_engine, read_parquet
+from .utils import schema_from_string
+
+__all__ = "read_delta_table"
+
+
+class DeltaTable:
+    """
+    Core DeltaTable Read Algorithm
+
+    Parameters
+    ----------
+    path: str
+        path of Delta table directory
+    version: int, default 0
+        DeltaTable Version, used for Time Travelling across the
+        different versions of the parquet datasets
+    columns: None or list(str)
+        Columns to load. If None, loads all.
+    engine : str, default 'auto'
+        Parquet reader library to use. Options include: 'auto', 'fastparquet',
+        'pyarrow', 'pyarrow-dataset', and 'pyarrow-legacy'. Defaults to 'auto',
+        which selects the FastParquetEngine if fastparquet is installed (and
+        ArrowDatasetEngine otherwise).  If 'pyarrow' or 'pyarrow-dataset' is
+        specified, the ArrowDatasetEngine (which leverages the pyarrow.dataset
+        API) will be used. If 'pyarrow-legacy' is specified, ArrowLegacyEngine
+        will be used (which leverages the pyarrow.parquet.ParquetDataset API).
+        NOTE: The 'pyarrow-legacy' option (ArrowLegacyEngine) is deprecated
+        for pyarrow>=5.
+    checkpoint: int, default None
+        DeltaTable protocol aggregates the json delta logs and creates parquet
+        checkpoint for every 10 versions.This will save some IO.
+        for example:
+            if you want to read 106th version of dataset, No need to read all
+            the all the 106 json files (which isIO heavy), instead read 100th
+            parquet checkpoint which contains all the logs of 100 versions in single
+            parquet file and now read the extra 6 json files to match the given
+            106th version.
+            (so IO reduced from 106 to 7)
+    storage_options : dict, default None
+        Key/value pairs to be passed on to the file-system backend, if any.
+    """
+
+    def __init__(
+        self,
+        path: str,
+        version: int = 0,
+        columns=None,
+        engine="auto",
+        checkpoint=None,
+        storage_options=None,
+    ):
+        self.path = str(path).rstrip("/")
+        self.version = version
+        self.pq_files = set()
+        self.delta_log_path = f"{self.path}/_delta_log"
+        self.fs, _, _ = get_fs_token_paths(path, storage_options=storage_options)
+        self.engine = engine
+        self.columns = columns
+        self.checkpoint = (
+            checkpoint if checkpoint is not None else self.get_checkpoint_id()
+        )
+        self.storage_options = storage_options
+        self.schema = None
+
+    def get_checkpoint_id(self):
+        """
+        if _last_checkpoint file exists, returns checkpoint_id else zero
+        """
+        try:
+            with self.fs.open(
```

Comment:
May be faster with `json.loads(self.fs.cat(fn))`

Reply:
Thanks for the suggestion, Now Changed it into fs.cat